### PR TITLE
[Network]az network firewall create: support multiple ip addresses for vhub firewall.

### PR DESCRIPTION
**Pull request by MyronFanQiu**
---

Fix Azure/azure-cli#13634

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S172

File: src/azure-firewall/azext_firewall/custom.py [hunk 1]
```diff
@@ -78,26 +90,43 @@ def create_azure_firewall(cmd, resource_group_name, azure_firewall_name, locatio
...
-        enable_dns_proxy if enable_dns_proxy is not None else False
-    firewall.additional_properties['Network.DNS.RequireProxyForNetworkRules'] = \
-        require_dns_proxy_for_network_rules if require_dns_proxy_for_network_rules is not None else True
-    firewall.additional_properties['Network.DNS.Servers'] = ','.join(dns_servers or '')
+    if sku is None or sku.lower() == 'azfw_vnet':
+        if firewall_policy is None:
+            firewall.additional_properties['Network.DNS.EnableProxy'] = \
+                enable_dns_proxy if enable_dns_proxy is not None else False
+            firewall.additional_properties['Network.DNS.RequireProxyForNetworkRules'] = \
+                require_dns_proxy_for_network_rules if require_dns_proxy_for_network_rules is not None else True
```
Comment (jsntcy) [inline src/azure-firewall/azext_firewall/custom.py:108]: default is true?

Comment (MyronFanQiu) [reply src/azure-firewall/azext_firewall/custom.py:108]: I think so. It's done by @haroldrandom. I invited you into a group chat for the context.

File: src/azure-firewall/azext_firewall/custom.py [hunk 2]
```diff
@@ -122,6 +151,29 @@ def update_azure_firewall(cmd, instance, tags=None, zones=None, private_ranges=N
         instance.additional_properties['Network.DNS.Servers'] = ','.join(dns_servers or '')
     if threat_intel_mode is not None:
         instance.threat_intel_mode = threat_intel_mode
+
+    if instance.hub_ip_addresses is None and hub_public_ip_addresses is not None:
```
Comment (jsntcy) [inline src/azure-firewall/azext_firewall/custom.py:155]: To make users easy to understand, can we tell users that they can only provide one of count (add) and ip_addresses (delete)? provide both help doc and verification logic.

File: src/azure-firewall/azext_firewall/custom.py [hunk 3]
```diff
@@ -122,6 +153,31 @@ def update_azure_firewall(cmd, instance, tags=None, zones=None, private_ranges=N
...
+            instance.hub_ip_addresses = HubIPAddresses(
+                public_ips=HubPublicIPAddresses(
+                    count=hub_public_ip_count
+                )
+            )
+
+    if hub_public_ip_addresses is not None:
+        try:
+            if len(hub_public_ip_addresses) > instance.hub_ip_addresses.public_ips.count:
+                raise CLIError('Number of public ip addresses must be less than existing ones.')
```
Comment (jsntcy) [inline src/azure-firewall/azext_firewall/custom.py:175]: less than or equal to

File: src/azure-firewall/azext_firewall/custom.py [hunk 4]
```diff
@@ -122,6 +153,31 @@ def update_azure_firewall(cmd, instance, tags=None, zones=None, private_ranges=N
...
+
+    if instance.hub_ip_addresses is None and hub_public_ip_addresses is not None:
+        raise CLIError('Cannot delete public ip addresses from vhub without creation.')
+    if hub_public_ip_count is not None:
+        try:
+            if instance.hub_ip_addresses.public_ips.count is not None and hub_public_ip_count > instance.hub_ip_addresses.public_ips.count:  # pylint: disable=line-too-long
+                instance.hub_ip_addresses.public_ips.count = hub_public_ip_count
+            else:
+                raise CLIError('Cannot decrease the count of hub ip addresses through --count.')
+        except AttributeError:
```
Comment (jsntcy) [inline src/azure-firewall/azext_firewall/custom.py:165]: When will this exception be raised?

Comment (MyronFanQiu) [reply src/azure-firewall/azext_firewall/custom.py:165]: if user doesn't set --count before and this filed would be None.

Comment (jsntcy) [reply src/azure-firewall/azext_firewall/custom.py:165]: Do you mean instance.hub_ip_address is None?

---
In reply to: [446598636](https://github.com/Azure/azure-cli-extensions/pull/1937#discussion_r446598636) [](ancestors = 446598636)

Comment (MyronFanQiu) [reply src/azure-firewall/azext_firewall/custom.py:165]: yes. Or `instance.hub_ip_address.public_ips` is None

**Pull request merged**